docs/source/traitsui_user_manual/testing/discussions/working_of_extensions.rst

```diff
@@ -0,0 +1,40 @@
+.. _testing-how-extension-works:
+
+How extension works
+===================
+
+|UITester| supports testing TraitsUI editors with various user interaction
+logic. However it is possible that projects' test code may require
+additional logic that is not supported by |UITester| by default. Furthermore,
+some projects may implement and maintain their own custom UI editors. Those
+custom UI editors are also by default not supported by |UITester|.
+
+The API allows extension such that
+
+* projects can test TraitsUI editors with user interactions that do not
+  come supported by default.
+* projects can reuse the testing API for testing custom editors.
+
+As described in :ref:`testing-add-new-interaction` and
+:ref:`testing-add-new-location`, the testing API can be extended by providing
+one or many registry objects. Internally, |UITester| maintains a list of
+registries ordered in decreasing priority. For example, if you provide multiple
+registries::
+
+    tester = UITester(registries=[custom_registry, another_registry])
+
+Interaction and location support registered in the first registry will
+supersede that of the second (if such implementation exists).
+
+This list of registries is passed on to all |UIWrapper| created from the
+|UITester|. When |UIWrapper.perform| or |UIWrapper.inspect| is called, the
+first registry that can support the given target and interaction will be used.
+Likewise, |UIWrapper.locate| follows the same rule for the given target and
+location. See :ref:`testing-target-interaction-location` for an explanation on
+these three types of objects.
+
+This is how |UITester| supports testing on TraitsUI editor: It always extends
+the given list of registries with more built-in registries that know how to
+interact with TraitsUI editors, and let |UIWrapper| do most of the work.
+
```

Comment:
We could also mention that this allows for overriding the default behavior.  This is implied by the content here but we may want to be explicit?  
Or we very well may not.  Overriding default behavior may be something we want to discourage.  I am not sure.  e.g. I can't think of a direct use case, but it is possible a user may have a different idea of what `DisplayedText()` should be or something?

Reply:
Yeah I agree there is an obvious use case and so we may not want to encourage it.  In the original documentation, there was a section about overriding the default behaviour, but really the motivation there was to explain the machinery about the registry priority.  Since that is reworded here, I think I was going with "this is implied here" so if someone really really wanted to override the default behaviour.

Reply:
That makes sense to me.  This is good as is in that case 👍